### DISCUSSION TITLE
FISH shell integration

### DISCUSSION THREAD

**Discussion by theMubashir919** [basecamp/omakub#304] [Q&A]
I love Omakub. Been using it for a while now and it just works. 
I recently switched from my default bash shell to Fish. Now I lost all the preconfigured bash setting from omakub. 
How do I move all the omakub bash configurations over to Fish shell? Someone kindly help with this...

**Comment by pedrotchang** [comment]
Go to your .bshrc file ask Claude.ai to convert to Fish shell and then add to config.fish.

**Comment by acramatte** [comment]
Fish shell is missing the `OMAKUB_PATH` env variable and the path to the `omakub` bin from: 
https://github.com/basecamp/omakub/blob/dbf627eeb2095477fca6ecfba4e9948f9b703cbb/defaults/bash/shell#L11

You can set the path with:
```bash
fish_add_path $HOME/.local/bin $HOME/.local/share/omakub/bin
```

and set the following variable in your `~/.config/fish/config.fish`
```
set -gx OMAKUB_PATH "$HOME/.local/share/omakub"
```

There might be more changes needed to the bash scripts under `$HOME/.local/share/omakub/bin/omakub-sub` but with the changes above you can already get back your themes, fonts, install and such.

**Comment by mhernandezve** [comment]
Bash configurations aren't compatible with Fish. The quickest way to configure Fish is using the `fish_config` command. If you have custom commands in bash, the best approach is to rewrite it from scratch. However, you could call the script using the following syntax `bash -c 'echo "This command runs in a new Bash shell"'` or by adding a shebang line at the beginning of the script. You can also use Fish plugins to extend the functionality of Fish. This is a good starting point for people who come from Bash https://fishshell.com/docs/current/tutorial.html#startup-where-s-bashrc
